### PR TITLE
Intram UK total by ITL 1 and 2

### PR DESCRIPTION
**Pull request by zorge69**
## Pull Request submission

Created two new outputs, intram_uk_itl1 and 2 aggregated at ITL level 1 and 2.
It's a UK total, containing BERD and NI data.
The code we are using for NI was not in the original mapper, so created a mapper itl_6162_ni.csv
Created two output schemas for exporting. Version is set to 0.7.1.

*Insert any instructions to help the reviewer, e.g. "install new requirements from `environment.yml`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/outputs/outputs_main.py [hunk 1]
```diff
@@ -170,6 +171,20 @@ def run_outputs(
...
+        output_intram_uk_itl_1_2(
+            outputs_df,
+            ni_full_responses,
+            config,
+            write_csv,
+            run_id,
+            postcode_mapper,
+            itl_mapper,
+        )
+        OutputMainLogger.info("Finished  Intram UK by ITL 1 and 2 output.")
```
Comment (JenCheshire) [inline src/outputs/outputs_main.py:186]: In the ITL1 output TLF is appearing twice (once as 'East Midlands' and once as  'East Midlands (England)').

TLH is also showing as just 'East' - is that right?

File: src/outputs/outputs_main.py [hunk 2]
```diff
@@ -170,6 +171,20 @@ def run_outputs(
...
+
```
Comment (JenCheshire) [inline src/outputs/outputs_main.py:187]: The totals (UK ITL 1 and UK ITL 2) are matching each other, and I tried to check against GB SAS + NI SAS and they were't tallying up.  Just had a quick look at the after the merge, and there are some postcodes not being mapped so possibly being dropped

**Review by JenCheshire — COMMENTED** [review]
I'll continue to review - just submitting these comments in case you want to start looking at them

File: src/outputs/outputs_main.py (hunk 1 above)
Comment (zorge69) [reply src/outputs/outputs_main.py:186]: 'East Midlands' and once as 'East Midlands (England)' - was an error in my mapper, fixed in Windows, needs fixing in DAP.

TLH Is 'East', it's OK

File: src/outputs/outputs_main.py (hunk 2 above)
Comment (zorge69) [reply src/outputs/outputs_main.py:187]: Good check. The sample on R: is not constructed, so some wrong codes are not mapped. This needs to be checked in DAP where  all postcodes should map to reg_code and itl

File: src/outputs/outputs_main.py (hunk 1 above)
Comment (zorge69) [reply src/outputs/outputs_main.py:186]: Updated the mapper in DAP

File: config/output_schemas/intram_uk_itl2.toml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+["Area Code (ITL3)"]
```
Comment (JenCheshire) [inline config/output_schemas/intram_uk_itl2.toml]: Should that be "Area Code(ITL2)" rather than ITL3?

File: src/outputs/intram_uk_itl_1_2.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
+"""The main file for the Intram by PG output."""
```
Comment (JenCheshire) [inline src/outputs/intram_uk_itl_1_2.py]: Update file description

File: config/output_schemas/intram_uk_itl2.toml (hunk 3 above)
Comment (zorge69) [reply config/output_schemas/intram_uk_itl2.toml]: Totally! Corrected.

File: src/outputs/intram_uk_itl_1_2.py (hunk 4 above)
Comment (zorge69) [reply src/outputs/intram_uk_itl_1_2.py]: Well spotted! Corrected.

**Review by JenCheshire — APPROVED** [review]

**Pull request merged**